job/jobcache.go

```diff
@@ -1,36 +1,86 @@
 package job
 
 import (
+	"os"
+	"os/signal"
 	"sync"
+	"time"
 )
 
-var (
-	AllJobs = &JobCache{
-		Jobs:   make(map[string]*Job),
-		rwLock: sync.Mutex{},
+type JobCache interface {
+	Get(id string) *Job
+	GetAll() map[string]*Job
+	Set(j *Job)
+	Delete(id string)
+	Persist() error
+}
+
+type MemoryJobCache struct {
+	Jobs            map[string]*Job
+	rwLock          sync.Mutex
+	jobDB           JobDB
+	persistWaitTime time.Duration
+}
+
+func NewMemoryJobCache(jobDB JobDB, persistWaitTime time.Duration) *MemoryJobCache {
+	if persistWaitTime == time.Duration(0) {
+		persistWaitTime = 5 * time.Second
 	}
-)
+	return &MemoryJobCache{
+		Jobs:            map[string]*Job{},
+		rwLock:          sync.Mutex{},
+		jobDB:           jobDB,
+		persistWaitTime: persistWaitTime,
+	}
+}
+
+func (c *MemoryJobCache) Init() {
```

Comment:
@ajvb so I mentioned it my previous comment about having objects ready to go upon construction, but this is doing a ton of things. A better name would be `Start()` for this.